domain/consensus/consensus.go

```diff
@@ -144,14 +144,6 @@ func (s *consensus) PruningPointAndItsAnticone() ([]*externalapi.DomainHash, err
 	return s.pruningManager.PruningPointAndItsAnticone()
 }
 
-// TODO: Remove this method once v3 is obsolete
-func (s *consensus) BlockWithTrustedData(blockHash *externalapi.DomainHash) (*externalapi.BlockWithTrustedData, error) {
-	s.lock.Lock()
-	defer s.lock.Unlock()
-
-	return s.pruningManager.BlockWithTrustedData(model.NewStagingArea(), blockHash)
```

Comment:
It seems like `pruningManager.BlockWithTrustedData` can be removed as well, currently it is never used. Is there any reason to keep it?